pkgs/top-level/all-packages.nix

```
@@ -25048,4 +25048,5 @@ in
 
   keycard-cli = callPackage ../tools/security/keycard-cli {};
 
+  streamlit = callPackage ../applications/science/machine-learning/streamlit { };
```

python2 is EOL in 2.5 months, just take care of this now.

Also, I would use the python3Packages.callPackage, it will be as if it was being installed from the python-modules directory
```suggestion
  streamlit = python3Packages.callPackage ../applications/science/machine-learning/streamlit { };
```

if you do this, then you will have to list the python packages individually in your expression.

---

For an example, docker-compose:

https://github.com/NixOS/nixpkgs/blob/7774945b70d1b57f2b7fc944e952119d33655b12/pkgs/top-level/all-packages.nix#L200
https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/virtualization/docker-compose/default.nix